vllm/distributed/parallel_state.py

```diff
@@ -342,6 +346,62 @@ def broadcast_object_list(self,
                                                 group=self.device_group)
         return obj_list
 
+    def send_object(self,
+                    obj: Any,
+                    dst: int,
+                    group: Optional[ProcessGroup] = None) -> None:
+        """Send the input object list to the destination rank."""
+        assert dst < self.world_size, f"Invalid dst rank ({dst})"
+
+        assert dst != self.rank, (
+            "Invalid destination rank. Destination rank is the same "
+            "as the current rank.")
+
+        current_device = _get_pg_default_device(group)
+        # Serialize object to tensor.
+        object_tensor, size_tensor = _object_to_tensor(obj, current_device,
+                                                       group)
+
+        # Send object size
+        torch.distributed.send(size_tensor, dst=dst, group=group)
```

Comment:
our api uses relative rank inside the group, but [pytorch](https://pytorch.org/docs/stable/distributed.html#torch.distributed.broadcast_object_list) uses global ranks:

> src ([int](https://docs.python.org/3/library/functions.html#int)) – Source rank from which to broadcast object_list. Source rank is based on global process group (regardless of group argument)

Reply:
Changed it so `{send,recv}`, `{send,recv}_object`, `{send,recv}_tensor_dict`. Also added new test for `{send,recv}`.